org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/PlexusContainerManager.java

```diff
@@ -86,7 +92,27 @@ public class PlexusContainerManager {
 
   private static final String PLEXUS_CORE_REALM = "plexus.core";
 
-  private static final ClassWorld CLASS_WORLD = new ClassWorld(PLEXUS_CORE_REALM, ClassWorld.class.getClassLoader());
+  private static final StackWalker STACK_WALKER = StackWalker.getInstance(Option.RETAIN_CLASS_REFERENCE);
+
+  private static final ClassWorld CLASS_WORLD = new ClassWorld(PLEXUS_CORE_REALM, ClassWorld.class.getClassLoader()) {
```

Comment:
Thinking a bit more about it, the soloution might be not to use one classworld for all containers, but an own classworld for each  container....

Reply:
Yes, I thought about that as well but assumed there is a reason to have only one class-world, probably less `ClassCastException`s.
But as long as objects created by Plug-in executions are not saved that should not be an issue.

Reply:
The reason was that I though it might not be necessary too have multiple "Worlds" but before we need to special hack the implementation it more seems there should be one world for each "maven container with extension" thing ...

Reply:
With one world per container there are definitely less changes necessary.
So lets see if that works. The extension tests passed. Lets see if all other tests pass as well.

Reply:
Looks good. All tests have passed.
So should we try it?